### PR TITLE
CampTix Tweaks: Fix keyboard accessibility of payment options

### PR DESCRIPTION
**Pull request by ryelle**
This PR changes the tab markup to use buttons, which natively catch keyboard actions. The "preferred payment option" label can be a button, since the input itself is hidden, and a toggle button matches the expected behavior better. All payment options are now output in the "Payment Options" list. If there is only one payment option, it is now just output as text, not a clickable button.

This also improves the labelling of the other payment options by wrapping them in a fieldset.

Fixes https://meta.trac.wordpress.org/ticket/4421

### Screenshots

Screen Shot 2019-12-20 at 3 51 57 PM
Screen Shot 2019-12-20 at 2 18 51 PM

### How to test the changes in this Pull Request:

1. Turn on both Stripe & PayPal in the payment settings
2. Go try to buy a ticket
3. Using just a keyboard, tab through the page to the end
4. Hit space/enter on "Other payment methods", you can now select either stripe or paypal
5. You can also shift-tab back up to the "Pay with Stripe" button, and hit space/enter to flip back to that.

Basically, everything you can click with a mouse should now be activated by the keyboard too.

### CODE REVIEW

File: public_html/wp-content/mu-plugins/camptix-tweaks/addons/class-payment-options.php [hunk 1]
```diff
@@ -201,41 +224,37 @@ private function only_one_payment_method( $payment_methods ) {
...
-		// Pre-select first payment method if stripe is not available
+		// Pre-select first payment method if stripe is not available.
 		if ( ! $this->is_stripe_available( $payment_methods ) && ! isset( $selected_payment_method ) ) {
 			$selected_payment_method = array_keys( $payment_methods )[0];
 		}
 
-		foreach ( $payment_methods as $payment_method_key => $payment_method ) {
-			if ( 'stripe' === $payment_method_key ) {
-				continue;
-			}
```
Comment (coreymckrill) [inline public_html/wp-content/mu-plugins/camptix-tweaks/addons/class-payment-options.php:220]: Why also include Stripe in the alternatives list when it already has its own button?

**Review by coreymckrill — COMMENTED** [review]
> If there is only one payment option, it is now just output as text, not a clickable button.

I noticed that if there are multiple payment options, but none of them are Stripe, the "Payment methods" heading is still a button, even though there is not a second one to toggle to.

File: public_html/wp-content/mu-plugins/camptix-tweaks/addons/class-payment-options.php (hunk 1 above)
Comment (ryelle) [reply public_html/wp-content/mu-plugins/camptix-tweaks/addons/class-payment-options.php:220]: Well, it's a list of all payment options, and it might not be immediately clear that you can click the stripe button again.

There's more potential for confusion for a screen reader user — reading through the list of "Payment Methods", it's more straightforward to select stripe from this list of options, vs knowing that they need to leave the list, and reverse-navigate back to the first toggle button.

**Comment by ryelle** [comment]
> I noticed that if there are multiple payment options, but none of them are Stripe, the "Payment methods" heading is still a button, even though there is not a second one to toggle to.

Good catch, fixed that in  https://github.com/WordPress/wordcamp.org/pull/304/commits/3b042a1cd3ece748f356bed48370c998f792ff98

File: public_html/wp-content/mu-plugins/camptix-tweaks/addons/class-payment-options.php (hunk 1 above)
Comment (coreymckrill) [reply public_html/wp-content/mu-plugins/camptix-tweaks/addons/class-payment-options.php:220]: That makes sense. What if we changed the label to "All payment methods" or something? It just seems confusing to me to see "other" and then have "Credit Card" be listed in there again.

Comment (ryelle) [reply public_html/wp-content/mu-plugins/camptix-tweaks/addons/class-payment-options.php:220]: "All payment methods" sounds good to me 👍

**Comment by ryelle** [comment]
Updated the label for "other payment methods" to "All", so it now looks like…

Screen Shot 2019-12-20 at 5 33 01 PM
Screen Shot 2019-12-20 at 5 33 11 PM

**Review by coreymckrill — APPROVED** [review]
